# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/rene/tetex/xputenv.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./texk/kpathsea/xputenv.c~	Tue Jan 26 21:49:35 1999
+++ ./texk/kpathsea/xputenv.c	Sun Jun 27 01:01:15 1999
@@ -26,6 +26,14 @@
 extern int putenv ();
 #endif /* not WIN32 */
 
+/* I don't understand the code here compleatly. My xdvi (and other programms)
+   dump core becouse saved_env_items[i] is set to var_name - but varname is
+   (at least when called from xputenv_int) not a malloced() variable. So it's
+   value at the next invocation is undefined. In my case the assertation for
+   old_item fired ...   I just define SMART_PUTENV here. I'm not sure if this
+   is the right thing to do - but it solves my problem   - Clifford */
+#define SMART_PUTENV
+
 /* This `x' function is different from the others in that it takes
    different parameters than the standard function; but I find it much
    more convenient to pass the variable and the value separately.  Also,
